helix-term/src/commands.rs

```diff
@@ -3284,33 +3286,88 @@ fn select_register(cx: &mut Context) {
     })
 }
 
-fn space_mode(cx: &mut Context) {
-    cx.on_next_key(move |cx, event| {
-        if let KeyEvent {
-            code: KeyCode::Char(ch),
-            ..
-        } = event
-        {
-            // TODO: temporarily show SPC in the mode list
-            match ch {
-                'f' => file_picker(cx),
-                'b' => buffer_picker(cx),
-                's' => symbol_picker(cx),
-                'w' => window_mode(cx),
-                'y' => yank_joined_to_clipboard(cx),
-                'Y' => yank_main_selection_to_clipboard(cx),
-                'p' => paste_clipboard_after(cx),
-                'P' => paste_clipboard_before(cx),
-                'R' => replace_selections_with_clipboard(cx),
-                // ' ' => toggle_alternate_buffer(cx),
-                // TODO: temporary since space mode took its old key
-                ' ' => keep_primary_selection(cx),
-                _ => (),
-            }
+macro_rules! mode_info {
+    // TODO: how to use one expr for both pat and expr?
+    // TODO: how to use replaced function name as str at compile time?
```

Comment:
Are you talking about https://doc.rust-lang.org/std/macro.stringify.html ?

Reply:
No, stringify only converts it to string but can't replace the `&str` at compile time.

Reply:
You can't do that without a proc macro.